core/src/com/cloud/agent/api/CheckS2SVpnConnectionsAnswer.java

```diff
@@ -76,4 +76,14 @@ public String getDetail(String ip) {
         }
         return null;
     }
+
+    public boolean isIPPresent(String ip) {
+        if (this.getResult()) {
+            Boolean status = ipToConnected.get(ip);
+            if (status != null) {
```

Comment:
Yes The check to see whether ip is there in the response. So it should be true even when status is false